code-challenges.test.js

```diff
@@ -47,21 +78,126 @@ const studyMinutesWeek2 = {
+
+  const l2G = (object) =>{
+  return Object.values(object).sort((a,b)=>a-b)
+  } 
+  
+  //TypeError: object.values is not a function
+  //capitalized object
+  // - Expected  - 1
+  //   + Received  + 1
+  //     Array [
+  //       10,
+  //   +   100,
+  //       15,
+  //       20,
+  //       45,
+  //       60,
+  //       65,
+  //   -   100,
+  //     ]
+  //added ((a,b)=>a-b) to sort in ascending order
+  //   Test Suites: 1 passed, 1 total
+  // Tests:       2 passed, 2 total
+
 // --------------------3) Create a function that takes in an array and returns an array of the accumulating sum. An empty array should return an empty array.
 
 // a) Create a test with expect statements for each of the variables provided.
 
+describe("totalSum", () => {
+  it("takes in an array and returns an array of the accumulating sum. An empty array should return an empty array.", () => {
 const accountTransactions1 = [100, -17, -23, -9]
-// Expected output: [100, 83, 60, 51]
-
+expectedOutput5 = [100, 83, 60, 51]
 const accountTransactions2 = [250, -89, 100, -96]
-// Expected output: [250, 161, 261, 165]
-
+expectedOutput6 = [250, 161, 261, 165]
 const accountTransactions3 = []
-// Expected output: []
+expectedOutput7 = []
+expect(totalSum(accountTransactions1)).toEqual(expectedOutput5)
+expect(totalSum(accountTransactions2)).toEqual(expectedOutput6)
+expect(totalSum(accountTransactions3)).toEqual(expectedOutput7)
+  })
+})
+
+//ReferenceError: totalSum is not defined
 
 // b) Create the function that makes the test pass.
+
+  //Create a function called totalSum.
+    //the parameter will be array.
+    //the function will have a for loop to check each value.
+    //the function will add each consecutive value at each index and will equal the array at that point with the index.
+    //the function will return an array with the sum added cumulatively
+  //run jest to verify function.
+
+// const totalSum =(array)=>{
+//   for(let i = 1 ; i < array.length ; i++) {
+//    sumTotal=0
+//    sum=[]
+//    sum.push(array[i]+sumTotal)
+   
+//   }
+//    return sum
+//   }
+
+  // - Expected  - 4
+  // + Received  + 1
+
+  //   Array [
+  // -   100,
+  // -   83,
+  // -   60,
+  // -   51,
+  // +   -9,
+  //   ]
+
+//changed sumTotal to array[i-1] to indicate last number and sum to be assigned to result of array[i]+array[i-1] for sum with push function to move to sumTotal array.
+
+// const totalSum =(array)=>{
+//   for(let i = 1 ; i < array.length ; i++) {
+//    sumTotal=[]
+//    sum= array[i] + array[i-1]
+//    sumTotal.push(sum)
+//   }
+//    return array
+//   }
+
+  // - Expected  - 3
+  // + Received  + 3
+
+  //   Array [
+  //     100,
+  // -   83,
+  // -   60,
+  // -   51,
+  // +   -17,
+  // +   -23,
+  // +   -9,
+  //   ]
+
+//changed the equation for array[i] = array[i] + array[i-1]. This is so at each index the value will be the result of the value of the current index plus the value of the last index.  
```

Comment:
Great work showing your progress on this problem!  Just make sure to watch your indentation.